### PR TITLE
Add GitHub action for SGX integration tests

### PR DESCRIPTION
**Pull request by rveerama1**
Currently it applies on pull request, once testing is done, I will add main branch requirements.

### CODE REVIEW

File: .github/workflows/integration-sgx.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - name: Code checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Run SGX integration tests
+        timeout-minutes: 60
+        run: scripts/dev_cli.sh tests --integration-sgx
+      - name: Run SGX integration tests for musl
+        timeout-minutes: 60
```
Comment (rbradford) [inline .github/workflows/integration-sgx.yaml:21]: Last time this ran the whole job took less than 10 minutes. I would set each of these to 10 minutes - we want to know if it's going to fail earlier.

File: .github/workflows/integration-sgx.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Cloud Hypervisor Tests (SGX)
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    if: github.event_name == 'pull_request'
```
Comment (rbradford) [inline .github/workflows/integration-sgx.yaml:9]: This is wrong - you've put "on: push" and then got an if check for pull_request...

Comment (rveerama1) [reply .github/workflows/integration-sgx.yaml:9]: oops

File: .github/workflows/integration-sgx.yaml (hunk 1 above)
Comment (rveerama1) [reply .github/workflows/integration-sgx.yaml:21]: I picked this value from Jenkinsfile

**Review by rbradford — APPROVED** [review]

**Pull request merged**